### DISCUSSION TITLE
Cache data between infinite query and sockets events

### DISCUSSION THREAD

**Discussion by tiagocorreiaalmeida** [TanStack/query#3649] [Q&A]
Hey, I'm trying to implement a mix of sockets and promised based calls into the same cache key.
Take for a example a chat channel, where once I enter it I need to call useInfiniteQuery() to get the latest 10 messages and as I scroll up I load more messages, other than loading the initial data trough a promise request I need to listen to events of messages being added, removed or updated.
My idea was to update the infinite query cache as those events come in but it becomes difficult and "expensive" to do so since the cache data of useInfinite query is stored as an array of arrays(data.pages[x]), any idea of what I could do in here?
If this is not something that react query is intend to support I will have to find other way to get there, thanks in advance

**Comment by TkDodo** [comment]
does this help?
https://tkdodo.eu/blog/using-web-sockets-with-react-query

**Comment by Zerebokep** [comment]
I'm currently on the same issue. I think the easiest way would be to write new messages from the subscription to data.pages[0]. @TkDodo please correct me if there is a better way.
